### ISSUE TITLE
export visibility from abc

### ISSUE THREAD

**Issue by kimuss** [Autodesk/maya-usd#3995]
**Describe the bug**
export does not read visibility channel

**Steps to reproduce**
1. import abc file with a visibility animation
2. usd export selection (or all) and check on visibility in export option
3. create stage from file just created
4. See the visibility does not have animation

**Expected behavior**
Use the visibility abc channel to write the visibility token on usd export, this work good on maya native element, like this:
![Screenshot 2024-11-12 172132](https://github.com/user-attachments/assets/60b6b7cd-ff6f-4818-be90-cd3432f79d8f)

**Specs (if applicable):**
 - Windows 11
 - Maya 2024.2
 - Maya USD plugin 0.30
 
 thank you

**Comment by santosg87** [comment]
thank you for reporting this! 

I can reproduce this as well. 

It seems the reason this is failing is due to how the alembic file is imported in maya. Since the animation is kept in the alembic node and not actually on the geometry itself. (when importing alembic, you can see an alembic node is created pointing to the file in disk). 


I logged and internal ticket and will bring it to the team to discuss any next steps.

**Comment by kimuss** [comment]
> 

>.. Since the animation is kept in the alembic node and not actually on the geometry itself.
> 

exactly, usd export seems not to consider the link of the alembic node, also with "channel" and "history" checked in the USD export option

thank you

**Comment by santosg87** [comment]
unfortunately, because there is a link, maya struggles with exporting that data, as the workflows on alembic are made that way to allow for updating the geometry cache if needed, so that node isn't exported. 

2 possible workarounds here:
1. bake the animation before exporting - that way you would have actual Keys that would translate on export.
2. a second option - although not very widely used, would be to just reference the alembic file directly on your USD stage, as USD does have some limited support to alembic.

at this time, this issue isn't a high priority, but we will keep it in our backlog for future improvements!

Cheers!

**Comment by BigRoy** [comment]
> at this time, this issue isn't a high priority, but we will keep it in our backlog for future improvements!

I'm a bit confused by this. I'd say this is an actual bug and points to potential bugs with other exports too. In essence it seems that Maya USD requires to find 'keyframes' as inputs for it to be exportable/animated instead of allowing arbitrary input connections to define a 'dynamic' result.

I'd argue, that as soon as something has an input connection that we should consider it animated (or at least if it's connected to something that maya in this 'optimization' logic has no knowledge of). 

To check this hypothesis - similar to the failing example described in the closed "duplicate" issue https://github.com/Autodesk/maya-usd/issues/4059 I've 'animated' the visibility with something else then keyframes, in this case - an expression. And indeed, the exporter is failing to export my "animated" result.

Basically, Maya USD export does not export what you see **without warning!** - which is just plain confusing behavior. Potentially an extra export argument that could enable a "assume input connection is animated" or alike which actually samples these connected attributes as if they were animated could be a common ground allowing both behaviors. However, I'd argue that that behavior should be the default behavior (and potentially the only behavior).

I assume it's mostly from an optimization point of view to ignore any `connection` to be treated as potentially animated?
However, if I do the same thing with e.g. expression to the `width` attribute on a polyCube's construction history with an expression - it exports just fine. So somehow, it does correctly identify the "geometry" changing (which I suppose would be a much heavier query?) even if not from animation curves. Why is doing the same on an attribute like `visibility` that much more problematic?

Here's the example scene:
[example_scene.zip](https://github.com/user-attachments/files/18260292/example_scene.zip)

**Comment by santosg87** [comment]
> > at this time, this issue isn't a high priority, but we will keep it in our backlog for future improvements!
> 
> I'm a bit confused by this. I'd say this is an actual bug and points to potential bugs with other exports too. In essence it seems that Maya USD requires to find 'keyframes' as inputs for it to be exportable/animated instead of allowing arbitrary input connections to define a 'dynamic' result.
> 
> I'd argue, that as soon as something has an input connection that we should consider it animated (or at least if it's connected to something that maya in this 'optimization' logic has no knowledge of).
> 
> To check this hypothesis - similar to the failing example described in the closed "duplicate" issue #4059 I've 'animated' the visibility with something else than keyframes, in this case - an expression. And indeed, the exporter is failing to export my "animated" result.

@BigRoy Sorry if my comment wasn't very clear!

I meant to say that we agree that this is indeed a bug. Given the other priorities that the team has right now, this issue is not on our currently scheduled work, but we might pick up in the future depending on the work we are doing. :)

although we see this on visibility for now, it makes one wonder if there are any other attributes that might behave similarly. - something for us to investigate a bit further as well
